src/CoreBundle/private/library/classes/TCMSListManager/TCMSListManagerEndPoint.class.php

```diff
@@ -301,7 +302,12 @@ public function GetPortalRestriction()
                 } else {
                     if ($portalMLTFieldExists) {
                         // mlt connection (record may be in many portals)
-                        $restriction .= " ($quotedMltTable.`target_id` IS NULL)";
+
+                        $mltSubSelect = "SELECT DISTINCT $quotedTableName.`id` FROM $quotedTableName";
```

Comment:
Can we move this code to a separate method (--> to simplify this method, avoid code duplication [these lines are almost identical to 289-294], and explain the purpose by method name)?

Reply:
I was thinking about it. But at least not easily:

It has a number of parameters (4?) and the difference between the two blocks would need another for a non-trivial if in that method.

Reply:
If we omit the `$quotedMltTable.target_id IN ($portalRestriction)` part, constructing $mltSubSelect is identical. So we can at least easily factor this path out, and the method only needs 3 arguments.

Reply:
done